x-pack/plugins/observability/public/pages/home/index.tsx

```diff
@@ -3,26 +3,23 @@
  * or more contributor license agreements. Licensed under the Elastic License;
  * you may not use this file except in compliance with the Elastic License.
  */
-import React, { useEffect } from 'react';
+import React, { useContext, useEffect } from 'react';
 import { useHistory } from 'react-router-dom';
-import { fetchHasData } from '../../data_handler';
-import { useFetcher } from '../../hooks/use_fetcher';
+import { ObsvSharedContext } from '../../context/shared_data';
 
 export function HomePage() {
   const history = useHistory();
-  const { data = {} } = useFetcher(() => fetchHasData(), []);
 
-  const values = Object.values(data);
-  const hasSomeData = values.length ? values.some((hasData) => hasData) : null;
+  const { sharedData } = useContext(ObsvSharedContext);
+
+  const { hasAnyData } = sharedData ?? {};
 
   useEffect(() => {
-    if (hasSomeData === true) {
+    if (hasAnyData) {
       history.push({ pathname: '/overview' });
     }
```

Comment:
Based on the refactors you did, this is kind of useless don't you think? I have a feeling that `hasAnyData` will never be filled with data at this point.